pkg/v1/tkg/test/tkgctl/tkgs_cc/tkgs_cc_workload_cluster_test.go

```diff
@@ -0,0 +1,93 @@
+// Copyright 2022 VMware, Inc. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package tkgs_cc
+
+import (
+	"bytes"
+	"io"
+	"os"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	"github.com/vmware-tanzu/tanzu-framework/pkg/v1/tkg/tkgctl"
+)
+
+const TKC_KIND = "kind: TanzuKubernetesCluster"
+
+var _ = Describe("TKGS Package Based LCM Beta - workload cluster tests", func() {
+	var (
+		clusterName string
+		namespace   string
+		err         error
+	)
+
+	BeforeEach(func() {
+		tkgctlClient, err = tkgctl.New(tkgctlOptions)
+		Expect(err).To(BeNil())
+	})
+
+	JustBeforeEach(func() {
+		err = tkgctlClient.CreateCluster(clusterOptions)
+	})
+
+	Context("when input file is cluster class based", func() {
+		BeforeEach(func() {
+			clusterName, namespace = ValidateClusterClassConfigFile(e2eConfig.WorkloadClusterOptions.ClusterClassFilePath)
+			e2eConfig.WorkloadClusterOptions.Namespace = namespace
+			e2eConfig.WorkloadClusterOptions.ClusterName = clusterName
+			deleteClusterOptions = getDeleteClustersOptions(e2eConfig)
+			clusterOptions.ClusterConfigFile = e2eConfig.WorkloadClusterOptions.ClusterClassFilePath
+			clusterOptions.ClusterName = e2eConfig.WorkloadClusterOptions.ClusterName
+			clusterOptions.Namespace = e2eConfig.WorkloadClusterOptions.Namespace
+		})
+
+		AfterEach(func() {
+			err = tkgctlClient.DeleteCluster(deleteClusterOptions)
+		})
+
+		It("should successfully create a cluster", func() {
+			Expect(err).ToNot(HaveOccurred())
+		})
+	})
+
+	Context("when input file is legacy cluster based", func() {
+		When("cluster create is invoked", func() {
+			AfterEach(func() {
+				err = tkgctlClient.DeleteCluster(deleteClusterOptions)
+			})
+
+			It("should successfully create a TKC workload cluster", func() {
+				Expect(err).ToNot(HaveOccurred())
+			})
+		})
+		When("Cluster create dry-run is invoked", func() {
+			var (
+				stdoutOld *os.File
+				r         *os.File
+				w         *os.File
+			)
+			BeforeEach(func() {
+				stdoutOld = os.Stdout
+				r, w, _ = os.Pipe()
+				os.Stdout = w
+
+				clusterOptions.GenerateOnly = true
+			})
+
+			It("should give TKC configuration as output", func() {
```

Comment:
its CClass file as input should get Cluster object as out put in dry-run mode right?

Reply:
done